Shared/Samples/Find closest facility to multiple points/FindClosestFacilityToMultiplePointsView.swift

```diff
@@ -0,0 +1,180 @@
+    @State private var error: Error? {
+        didSet { errorAlertIsShowing = error != nil }
+    }
+    
+    var body: some View {
+        MapView(
+            map: model.map,
+            graphicsOverlays: [model.facilityGraphicsOverlay, model.incidentGraphicsOverlay]
+        )
+        .onSingleTapGesture { _, mapPoint in
+            tapLocation = mapPoint
+        }
+        .task(id: tapLocation) {
+            // Add an incident at the tap location.
+            guard let tapLocation else { return }
+            
+            do {
+                try await model.updateIncident(to: tapLocation)
+            } catch {
+                self.error = error
+            }
+        }
+        .task {
+            // Create the default parameters from the task when the sample loads.
+            do {
+                try await model.closestFacilityParameters = model.closestFacilityTask.makeDefaultParameters()
+                model.closestFacilityParameters?.setFacilities(model.facilities)
+            } catch {
+                self.error = error
+            }
+        }
+        .alert(isPresented: $errorAlertIsShowing, presentingError: error)
+    }
+}
+
+private extension FindClosestFacilityToMultiplePointsView {
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A map with a streets basemap centered on San Diego, CA, USA.
+        let map = {
+            let map = Map(basemapStyle: .arcGISStreets)
+            map.initialViewpoint = Viewpoint(latitude: 32.727, longitude: -117.175, scale: 144_400)
+            return map
+        }()
+        
+        /// The graphics overlay for the incident graphics.
+        let incidentGraphicsOverlay = GraphicsOverlay()
+        
+        /// The graphics overlay for the facility graphics.
+        let facilityGraphicsOverlay = GraphicsOverlay()
+        
+        /// The task for finding the closest facility.
+        let closestFacilityTask = ClosestFacilityTask(url: .sanDiegoNetworkAnalysis)
+        
+        /// The parameters to be passed to the task to find the closest facility.
+        var closestFacilityParameters: ClosestFacilityParameters?
+        
+        /// A list of facilities around San Diego, CA, USA.
+        let facilities = {
+            let facilityPoints = [
+                Point(x: -13_042_130, y: 3_860_128, spatialReference: .webMercator),
+                Point(x: -13_042_193, y: 3_862_449, spatialReference: .webMercator),
+                Point(x: -13_046_883, y: 3_862_705, spatialReference: .webMercator),
+                Point(x: -13_040_540, y: 3_862_925, spatialReference: .webMercator),
+                Point(x: -13_042_571, y: 3_858_982, spatialReference: .webMercator),
+                Point(x: -13_039_785, y: 3_856_693, spatialReference: .webMercator),
+                Point(x: -13_049_024, y: 3_861_994, spatialReference: .webMercator)
+            ]
+            return facilityPoints.map { Facility(point: $0) }
```

Comment:
```suggestion
            return facilityPoints.map(Facility.init(point:))
```
A fancier way to do it

Reply:
Is one preferred over the other?

Reply:
Would this be better as a computed static property too?

Reply:
I've no strong opinion on either. The array is quite small so computed or not doesn't make a huge difference.

Reply:
> Is one preferred over the other?

Yes. The non-closure way is preferred because it is more efficient. With `map { Facility(point: $0) }`, the `transform` function is a closure whose body calls the `Facility` initializer. With `map(Facility.init(point:))`, the `transform` function is the initializer itself. In other words, it eliminates the unnecessary wrapping closure.

Reply:
Cool! Thanks philium!